src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -103,25 +107,107 @@ final class ProxyGenerator extends ClassWriter {
     private static final ProxyMethod equalsMethod;
     private static final ProxyMethod toStringMethod;
 
+    private static final ClassModel TEMPLATE;
+
+    private static final Utf8Entry UE_Method;
+    private static final MethodRefEntry MRE_Class_getClassLoader;
+    private static final MethodRefEntry MRE_NoSuchMethodError_init;
+    private static final ClassEntry CE_NoSuchMethodError;
+    private static final FieldRefEntry FRE_Proxy_h;
+    private static final ClassEntry CE_NoClassDefFoundError;
+    private static final MethodRefEntry MRE_NoClassDefFoundError_init;
+    private static final MethodRefEntry MRE_Throwable_getMessage;
+    private static final MethodRefEntry MRE_Class_forName;
+    private static final ClassEntry CE_Class;
+    private static final MethodRefEntry MRE_Class_getMethod;
+    private static final ClassEntry CE_Object;
+    private static final InterfaceMethodRefEntry IMRE_InvocationHandler_invoke;
+    private static final MethodRefEntry MRE_UndeclaredThrowableException_init;
+    private static final ClassEntry CE_UndeclaredThrowableException;
+    private static final ClassEntry CE_Throwable;
+    private static final ClassEntry CE_NoSuchMethodException;
+    private static final ClassEntry CE_ClassNotFoundException;
+    private static final List<StackMapFrameInfo.VerificationTypeInfo> THROWABLE_STACK;
+
     static {
+        var cc = ClassFile.of();
+        var entries = new ArrayList<PoolEntry>(20);
+        var q = new Object() {
+            PoolEntry[] entries;
+            int i;
+            @SuppressWarnings("unchecked")
+            <T extends PoolEntry> T next() {
+                return (T) TEMPLATE.constantPool().entryByIndex(entries[i++].index());
+            }
+        };
+        TEMPLATE = cc.parse(cc.build(CD_Proxy, clb -> {
+            clb.withSuperclass(CD_Proxy);
+            generateConstructor(clb);
+            generateLookupAccessor(clb);
+            var cp = clb.constantPool();
+            q.entries = new PoolEntry[] {
```

Comment:
line 174-197 must match the order of CP entries being added.

Is there other way to avoid `q.next()` coupling with the order of CP entries in `q.entries`?  Maybe define constants for the indices to `q.entries` such that writing and reading the CP entries from the array using the constant index is easier to read and less error-prone?

Reply:
I've rewritten the initialization to use indexes inside an array.
Thanks.